src/pages/Facility/settings/devices/components/DeviceServiceHistory.tsx

```diff
@@ -0,0 +1,132 @@
+  const { t } = useTranslation();
+  const [isServiceSheetOpen, setIsServiceSheetOpen] = useState(false);
+  const [selectedServiceHistory, setSelectedServiceHistory] =
+    useState<ServiceHistoryResponse | null>(null);
+
+  const {
+    data: serviceHistory,
+    isLoading,
+    refetch,
+  } = useQuery({
+    queryKey: ["device", facilityId, deviceId, "serviceHistory"],
+    queryFn: query(deviceApi.serviceHistory.list, {
+      pathParams: {
+        facility_external_id: facilityId,
+        device_external_id: deviceId,
+      },
+    }),
+  });
+
+  const handleAddService = () => {
+    setSelectedServiceHistory(null);
+    setIsServiceSheetOpen(true);
+  };
+
+  const handleEditService = (service: ServiceHistoryResponse) => {
+    setSelectedServiceHistory(service);
+    setIsServiceSheetOpen(true);
+  };
+
+  const handleServiceSheetClose = () => {
+    setIsServiceSheetOpen(false);
+    refetch();
+  };
+
+  return (
+    <Card>
+      <CardHeader className="flex flex-row items-center justify-between">
+        <div>
+          <CardTitle>{t("service_history")}</CardTitle>
+        </div>
+        <Button onClick={handleAddService} size="sm">
+          <Plus className="h-4 w-4 mr-2" />
+          {t("service_record_add")}
+        </Button>
+      </CardHeader>
+      <CardContent>
+        {isLoading ? (
+          <div className="space-y-2">
+            <Skeleton className="h-10 w-full" />
+            <Skeleton className="h-10 w-full" />
+            <Skeleton className="h-10 w-full" />
+          </div>
+        ) : serviceHistory && serviceHistory?.results?.length > 0 ? (
+          <Table>
+            <TableHeader>
+              <TableRow>
+                <TableHead>{t("service_date")}</TableHead>
+                <TableHead>{t("service_notes")}</TableHead>
+                <TableHead className="text-right">{t("actions")}</TableHead>
+              </TableRow>
+            </TableHeader>
+            <TableBody>
+              {serviceHistory.results.map((service: ServiceHistoryResponse) => (
+                <TableRow key={service.id}>
+                  <TableCell className="font-medium">
+                    {format(new Date(service.serviced_on), "PPP")}
+                  </TableCell>
+                  <TableCell className="max-w-md truncate">
+                    {service.note}
+                  </TableCell>
+                  <TableCell className="text-right">
+                    <Button
+                      variant="ghost"
+                      size="icon"
+                      onClick={() => handleEditService(service)}
+                    >
+                      <Edit className="h-4 w-4" />
+                    </Button>
+                  </TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+        ) : (
+          <div className="text-center py-6 text-muted-foreground">
+            {t("service_records_none", "No service history available")}
```

Comment:
why pass a second argument? 🤔 
```suggestion
            {t("service_records_none")}
```

Reply:
Same here. Make the changes suggested.

Reply:
I was actually working on it and made some changes locally. I thought I would push the changes first, then request a review